config/base.go

```diff
@@ -0,0 +1,92 @@
+package config
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/notaryproject/notation-go/dir"
+)
+
+var (
+	// ConfigPath is the path for config.json
+	ConfigPath string
+	// SigningKeysPath is the path for signingkeys.json
+	SigningKeysPath string
+
+	// configInfo is the information of config.json
+	configInfo     *Config
+	configInfoOnce sync.Once
+
+	// signingKeysInfo is the information of signingkeys.json
+	signingKeysInfo     *SigningKeys
+	signingKeysInfoOnce sync.Once
+)
+
+func init() {
+	ConfigPath = dir.Path.Config()
+	SigningKeysPath = dir.Path.SigningKeyConfig()
+}
+
+// Configuration is the main config struct of notation-go
+type Configuration struct {
+	*Config
+	*SigningKeys
+}
```

Comment:
Accepted your suggestion, however, by using pointer can reduce once copy because Config and SigningKeys are loaded separately and combined together, which needs a copy or passing a pointer.

Reply:
Removed the Configuration structure